workspace/__ardulib__/Graphics/XGraphics.h

```diff
@@ -0,0 +1,458 @@
+    return res;
+}
+
+class Stroke : public XBitmap {
+private:
+    XColor _color = { 0, 0, 0 };
+
+public:
+    Stroke(XBitmap* parent, XColor color) {
+        _parent = parent;
+        _color = color;
+    }
+
+    void setParams(XColor color);
+    const XColor* getStrokeColor() const;
+};
+
+const XColor* Stroke::getStrokeColor() const {
+    return &_color;
+}
+
+void Stroke::setParams(XColor color) {
+    _color = color;
+}
+
+class FillSolid : public XBitmap {
+private:
+    XFill* _fill = nullptr;
+
+public:
+    FillSolid(XBitmap* parent, XColor color) {
+        _fill = new XSolidFill(color);
+        _parent = parent;
+    }
+
+    void setParams(XColor color);
+    const XFill* getFill() const;
+};
+
+void FillSolid::setParams(XColor color) {
+    _fill->setSolidColor(color);
+}
+
+const XFill* FillSolid::getFill() const {
+    return _fill;
+}
+
+class Point : public XBitmap {
+private:
+    XVector2<int16_t> _A;
+    XVector2<int16_t> _old_A;
+
+public:
+    Point(XBitmap* parent, int16_t x, int16_t y) {
+        _parent = parent;
+         _A = XVector2<int16_t>(x, y);
+        _old_A = _A;
+    }
+
+    void setParams(int16_t x, int16_t y);
+    bool renderBitmapLine(XRender* renderer, int16_t line, uint16_t* buffer);
+};
+
+void Point::setParams(int16_t x, int16_t y) {
+    _old_A = _A;
+    _A = XVector2<int16_t>(x, y);
+}
+
+bool Point::renderBitmapLine(XRender* renderer, int16_t line, uint16_t* buffer) {
+    bool res = false;
+
+    if (getStrokeColor() != nullptr) {
+
+        if (line == _A.y) {
+
+            uint16_t color = color888to565(getStrokeColor()->r, getStrokeColor()->g, getStrokeColor()->b);
+            buffer[_A.x] = color;
+            res = true;
+        }
+
+        if (line == _old_A.y) {
+            res = true;
+        }
+    }
+
+    return res;
+}
+
+class Line : public XBitmap {
+private:
+    XVector2<int16_t> _A;
+    XVector2<int16_t> _B;
+    XVector2<int16_t> _old_A;
+    XVector2<int16_t> _old_B;
+
+public:
+    Line(XBitmap* parent, int16_t x0, int16_t y0, int16_t x1, int16_t y1) {
+        _parent = parent;
+        _A  = XVector2<int16_t>(x0, y0);
+        _B  = XVector2<int16_t>(x1, y1);
+        _old_A = _A;
+        _old_B = _B;
+    }
+
+    void setParams(int16_t x0, int16_t y0, int16_t x1, int16_t y1);
+    bool renderBitmapLine(XRender* renderer, int16_t line, uint16_t* buffer);
+};
+
+void Line::setParams(int16_t x0, int16_t y0, int16_t x1, int16_t y1) {
+    _old_B = _B;
+    _old_A = _A;
+    _A = XVector2<int16_t>(x0, y0);
+    _B = XVector2<int16_t>(x1, y1);
+}
+
+bool Line::renderBitmapLine(XRender* renderer, int16_t line, uint16_t* buffer) {
+    bool res = false;
+
+    if (getStrokeColor() != nullptr) {
+
+        if (_old_A.y == _old_B.y) {
+            if (line == _old_A.y) {
+                res = true;
+            }
+        }
+
+        if (_old_A.y != _old_B.y) {
+            int16_t old_minY;
+            int16_t old_maxY;
+            if (_old_A.y > _old_B.y) {
```

Comment:
Looks like such blocks can be easily replaced with the usage of `max` and `min` functions.
Two lines instead of seven 🤓